cmd/skaffold/app/cmd/config/util.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2018 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/context"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	yaml "gopkg.in/yaml.v2"
+)
+
+const defaultConfigLocation = ".skaffold/config"
+
+func resolveKubectlContext() {
+	if kubecontext != "" {
+		return
+	}
+
+	context, err := context.CurrentContext()
+	if err != nil {
+		logrus.Warn(errors.Wrap(err, "retrieving current kubectl context"))
+		kubecontext = "default"
```

Comment:
add this as a constant somewhere?

Reply:
actually I reworked this logic to fall back to the global values (not a "default" context) if one isn't set, thanks for pointing this out. updated